### PR TITLE
Mixed fixes for eepromutils

### PR DESCRIPTION
**Pull request by nicola-lunghi**
Mixed fixes for eepromutils:
- fix missing headers
- add the possibility to select eeprom address and bus
- added .gitignore to ignore compiled files

### CODE REVIEW

File: eepromutils/eepflash.sh [hunk 1]
```diff
@@ -137,3 +158,7 @@ if [ $rc != 0 ]; then
 else
 	echo "Done."
 fi
+
+echo "Closing EEPROM Device."
+echo "0x$ADDR" > $SYS/delete_device
+
```
Comment (pelwell) [inline eepromutils/eepflash.sh]: Spurious blank line.

**Comment by pelwell** [comment]
1. eepmake compiles without errors on the two platforms I use regularly - what declaration is required from unistd.h?

2. I understand why being able to select a bus might be useful, but what is the purpose of providing an alternate device address? The firmware will only read from address 0x50, so appearing to support alternatives might be confusing.

**Comment by nicola-lunghi** [comment]
1. eepmake compiles without errors on the two platforms I use regularly - what declaration is required from unistd.h?

```
$ make 
cc eepmake.c -o eepmake -Wno-format
eepmake.c: In function ‘parse_command’:
eepmake.c:211:21: warning: implicit declaration of function ‘read’ [-Wimplicit-function-declaration]
    ssize_t result = read(random_file, &vinf->serial_1, 16);
                     ^~~~
eepmake.c:212:4: warning: implicit declaration of function ‘close’ [-Wimplicit-function-declaration]
    close(random_file);
    ^~~~~
cc eepdump.c -o eepdump -Wno-format
```
```
$ gcc --version
gcc (Ubuntu 6.4.0-8ubuntu1) 6.4.0 20171010
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
```
$ man close 6

CLOSE(2)                                                                                                Linux Programmer's Manual                                                                                               CLOSE(2)

NAME
       close - close a file descriptor

SYNOPSIS
       #include <unistd.h>

       int close(int fd);

DESCRIPTION
       close()  closes  a file desc
```
2. I understand why being able to select a bus might be useful, but what is the purpose of providing an alternate device address? The firmware will only read from address 0x50, so appearing to support alternatives might be confusing.

I have multiple EEPROM connected to the same i2c channel (8) with different addresses and I launch the script to program them sequentially from another script.
the script will default to 0x50 if not used

File: eepromutils/eepflash.sh (hunk 1 above)
Comment (nicola-lunghi) [reply eepromutils/eepflash.sh]: Thanks

**Comment by pelwell** [comment]
I can't see a reason to use open() over fopen(), so a better fix would be to use stdio functions throughout.

**Comment by nicola-lunghi** [comment]
> I can't see a reason to use open() over fopen(), so a better fix would be to use stdio functions throughout.

I agree with you.
But there are also other part of this file that needs rewrite not only that.
I was thinking of rewriting this from scratch in Python but for now no time to do that sorry.

**Pull request merged**